src/pages/transactions/_components/TransactionRow.tsx

```diff
@@ -0,0 +1,195 @@
+import { FiFileText } from "react-icons/fi";
+import { TransactionsIcon } from "@components/icons/Transactions";
+import { NumericFormat } from "react-number-format";
+import BigNumber from "bignumber.js";
+import { ConfirmCheck } from "@components/icons/ConfirmCheck";
+import { secondsToDhmsDisplay } from "@components/helpers/DurationHelper";
+import { Link } from "@components/commons/Link";
+import { RejectedCross } from "@components/icons/RejectedCross";
+import clsx from "clsx";
+import {
+  TransactionI,
+  TransactionStatus,
+  TransactionType,
+} from "./TransactionData";
+
+const iconMapping = {
+  [TransactionType.ContractCall]: FiFileText,
+  [TransactionType.Transaction]: TransactionsIcon,
+};
+
+export default function TransactionRow({ data }: { data: TransactionI }) {
+  const Icon = iconMapping[data.transactionType];
+  return (
+    <div>
+      {/* for desktop and tablet */}
+      <div className="hidden md:block">
+        <div className="grid grid-cols-8 lg:grid-cols-12 gap-5 py-3.5 lg:py-4">
+          <div className="col-start-1 col-end-3">
+            <div className="flex flex-row">
+              <Icon size={24} className="text-white-50 stroke-white-50" />
+              <div className="flex flex-col overflow-hidden ml-2 lg:ml-4 text-base">
+                <span className="text-white-50">{data.transactionType}</span>
+              </div>
+            </div>
+            <TransactionLinkRow
+              label="Hash:"
+              pathname={`/transactions/${data.hash}`}
+              value={data.hash}
```

Comment:
lyka added a `truncateTextFromMiddle` function, let's use that, similar for addresses